packages/amplify-cli/src/commands/diagnose.ts

```diff
@@ -87,9 +87,16 @@ const showLearnMore = (showOptOut: boolean): void => {
 const zipSend = async (context: Context, skipPrompts: boolean, error: Error | undefined): Promise<void> => {
   const choices = ['Generate report', 'Nothing'];
   if (!skipPrompts) {
-    const diagnoseAction = await prompter.pick('What would you like to do?', choices);
-    if (diagnoseAction !== choices[0]) {
-      return;
+    if (DebugConfig.Instance.promptSendReport()) {
+      const result = await prompter.yesOrNo('Help improve Amplify CLI by sharing non sensitive configurations on failures', false);
+      DebugConfig.Instance.setAndWriteShareProject(result);
+      if (result === false) {
+        return;
+      }
+      const diagnoseAction = await prompter.pick('What would you like to do?', choices);
+      if (diagnoseAction !== choices[0]) {
+        return;
+      }
```

Comment:
We should.

1. Double check this condition with https://github.com/aws-amplify/amplify-cli/blob/120e7feba8655f4404da756b0183c38f70ce3f95/packages/amplify-cli/src/commands/diagnose.ts#L35 . I.e. we must never ask this question in headless mode.
2. We should undo all the changes made to e2e/migration/other tests. They don't scale. Current implementation covers very narrow failure modes that are expected. All unexpected errors will now result in `nexpect` timing out on unexpected prompt should failure happen (transient or not). This will contribute to overall e2e test runtime.
3. Instead of that we should.
    1. Opt out existing e2e tests by default, i.e. make sure `DebugConfig.Instance.promptSendReport()` returns `false`, so this prompt is never asked there.
    2. Add test(s) and test utils dedicated to covering that particular scenario.